### PR TITLE
mimirsbrunn: set empty poi name with type

### PR DESCRIPTION
**Pull request by patochectp**

### CODE REVIEW

File: src/osm_reader/poi.rs [hunk 1]
```diff
@@ -96,19 +96,14 @@ fn parse_poi(osmobj: &osmpbfreader::OsmObj,
         }
     };
 
-    let name = osmobj.tags().get("name").map_or("", |name| name);
+    let name = osmobj.tags().get("name").map_or(osmobj.tags().get("amenity").unwrap(), |name| name );
```
Comment (TeXitoi) [inline src/osm_reader/poi.rs]: You don't want a hard unwrap. imagine a poi with `tags =[("leisure", "park")]`, kaboom.

Comment (patochectp) [inline src/osm_reader/poi.rs]: done

File: src/osm_reader/poi.rs [hunk 2]
```diff
@@ -96,23 +96,28 @@ fn parse_poi(osmobj: &osmpbfreader::OsmObj,
...
     if coord.is_default() {
         info!("The poi {} is rejected, cause: could not compute coordinates.",
               id);
         return None;
     }
 
+    if name.is_none() {
+        info!("The poi {} is rejected, cause: could not compute name.", id);
+        return None;
+    }
```
Comment (TeXitoi) [inline src/osm_reader/poi.rs]:
```rust
let name = if let Some(s) = name {
    s
} else {
    info!("The poi {} is rejected, cause: could not compute name.", id);
    return None;
};
```
and remove the `unwrap()` after.

File: src/osm_reader/poi.rs [hunk 3]
```diff
@@ -96,15 +96,15 @@ fn parse_poi(osmobj: &osmpbfreader::OsmObj,
         }
     };
 
-    let name = osmobj.tags().get("name").map_or("", |name| name);
+    let name = osmobj.tags().get("name").map_or(osmobj.tags().get("amenity"), |name| Some(name) );
```
Comment (TeXitoi) [inline src/osm_reader/poi.rs]: extra space at the end make rust-fmt test fail.

**Comment by patochectp** [comment]
done

File: tests/bragi_test.rs [hunk 4]
```diff
@@ -382,10 +382,10 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
     let types = get_types(&geocodings);
     assert_eq!(count_types(&types, "poi"), 1);
 
-    // we search for a POI (id = 663715752) with a label but an empty name, it should be filtered)
-    let geocodings = get_results(bragi_get("/autocomplete?q=(Saint-Martin-d'Hères)"));
+    // we search for POI with a label but an empty name set with type)
```
Comment (antoine-de) [inline tests/bragi_test.rs]: proposal for a clearer comment:

// we search for POIs with a type but an empty name, we should have set the name with the type.
// for exemple there are parkings without name (but with the tag "anemity" = "Parking"), we should be able to query them

**Pull request merged**